### ISSUE TITLE
We should have a snippet that inserts all required-to-be-initialized fields

### ISSUE THREAD

**Issue by jasonmalinowski** [dotnet/roslyn#61550]
With [required members](https://github.com/dotnet/csharplang/blob/main/proposals/required-members.md), there is a certain list of members you may need to initialize with an object initializer. We should have some snippet that inserts all of those, and lets the user tab between all of them. We could also fill in default values the same way we do argument completion.

The only real issue I see (and don't have an answer I like) is what the trigger looks like. We could stick an extra special placeholder in completion to insert all of them, or have some other semantic snippet that's just a well known thing you type. But I don't like either of those...

**Comment by sharwell** [comment]
A refactoring (and/or code fix for the compiler error) could be sufficient here, similar to what we do for switch statements.

**Comment by CyrusNajmabadi** [comment]
Yeah.  I'd far prefer a lightbulb-action here for this.  It could be driven off of the diagnostic the compiler will give.  Or we could just write the refactoring *very* simply.

**Comment by jasonmalinowski** [comment]
@CyrusNajmabadi @sharwell True, we should have a fix either way, to make it easier to update existing code. I guess as long as we ensure that is high enough priority then it means typing `new C { }` and then immediately invoking Ctrl + . between the {}s will pick this one. Otherwise it'd seem a bit unfortunate if you have to really find the fix.

**Comment by sharwell** [comment]
The code fix would be for a compiler error, which is the highest priority default bucket.

**Comment by CyrusNajmabadi** [comment]
Closing out due to lack of feedback.
